rskj-core/src/main/java/org/ethereum/core/BlockHeaderBuilder.java

```diff
@@ -0,0 +1,300 @@
+        this.useRskip92Encoding = useRskip92Encoding;
+        return this;
+    }
+
+    public BlockHeaderBuilder setIncludeForkDetectionData(boolean includeForkDetectionData) {
+        this.includeForkDetectionData = includeForkDetectionData;
+        return this;
+    }
+
+    public BlockHeaderBuilder setParentHashFromKeccak256(Keccak256 parentHash) {
+        this.parentHash = copy(parentHash);
+        return this;
+    }
+
+    public BlockHeaderBuilder setParentHash(byte[] parentHash) {
+        this.parentHash = copy(parentHash);
+        return this;
+    }
+
+    public BlockHeaderBuilder setEmptyUnclesHash() {
+        this.unclesHash = EMPTY_LIST_HASH;
+        return this;
+    }
+
+    public BlockHeaderBuilder setUnclesHash(byte[] unclesHash) {
+        this.unclesHash = copy(unclesHash);
+        return this;
+    }
+
+    public BlockHeaderBuilder setCoinbase(RskAddress coinbase) {
+        this.coinbase = coinbase;
+        return this;
+    }
+
+    public BlockHeaderBuilder setDifficultyFromBytes(@Nullable byte[] data) {
+        // This is to make it compatible with RLP.parseBlockDifficulty() which was previously
+        // user (but I think it was wrongly included in the RLP class, because these arguments
+        // do not come from any RLP parsing).
+        if (data != null) {
+            difficulty = new BlockDifficulty(new BigInteger(data));
+        } else {
+            difficulty = null;
+        }
+        return this;
+    }
+
+    public BlockHeaderBuilder setEmptyMergedMiningForkDetectionData() {
+        mergedMiningForkDetectionData = new byte[12];
+        return this;
+    }
+
+    public BlockHeaderBuilder setEmptyExtraData() {
+        extraData = new byte[]{};
+        return this;
+    }
+
+    public BlockHeaderBuilder setEmptyLogsBloom() {
+        logsBloom = copy(new Bloom().getData());
+        return this;
+    }
+
+    public BlockHeaderBuilder setEmptyStateRoot() {
+        stateRoot = EMPTY_TRIE_HASH;
+        return this;
+    }
+
+    public BlockHeaderBuilder setEmptyReceiptTrieRoot() {
+        receiptTrieRoot = EMPTY_TRIE_HASH;
+        return this;
+    }
+
+    private void normalize() {
+        extraData = normalizeValue(extraData, new byte[0]);
+        bitcoinMergedMiningHeader = normalizeValue(bitcoinMergedMiningHeader, new byte[0]);
+        bitcoinMergedMiningMerkleProof = normalizeValue(bitcoinMergedMiningMerkleProof, new byte[0]);
+        bitcoinMergedMiningCoinbaseTransaction = normalizeValue(bitcoinMergedMiningCoinbaseTransaction, new byte[0]);
+
+        unclesHash = normalizeValue(unclesHash, EMPTY_LIST_HASH);
+        coinbase = normalizeValue(coinbase, RskAddress.nullAddress());
+        stateRoot = normalizeValue(stateRoot, EMPTY_TRIE_HASH);
+        txTrieRoot = normalizeValue(txTrieRoot, EMPTY_TRIE_HASH);
+        receiptTrieRoot = normalizeValue(receiptTrieRoot, EMPTY_TRIE_HASH);
+        logsBloom = normalizeValue(logsBloom, new Bloom().getData());
```

Comment:
Before this builder, `logsBloom` used to be initialized by default with empty byte arrays of length 0 or 32 which are not valid values. That is why from now on `new Bloom().getData()` (aka `new byte[256]`) is used